### PR TITLE
Issue #5894: Remove lambda parameters processing

### PR DESCRIPTION
**Pull request by kazachka**
Issue #5894
Remove processing of lambda from ParameterNameCheck, tests added.

### CODE REVIEW

**Comment by rnveach** [comment]
@romani appveyor failed while travis passed for code coverage. Is there a possible issue with travis?

When comparing the 2, I see appveyor doesn't run `clean`.
After the 2742 tests are run, travis runs the following:
````
restore-instrumented-classes (default-cli)
report (default-report)
Analyzed bundle 'checkstyle' with 396 classes
````
while appveyor runs 
````
restore-instrumented-classes (default-restore-instrumented-classes)
maven-jar-plugin:3.1.0:jar (default-jar)
maven-jar-plugin:3.1.0:test-jar (default)
maven-failsafe-plugin:2.22.0:integration-test (integration-test)
IT tests.
maven-checkstyle-plugin:3.0.0:check (sevntu-checkstyle-check)
jacoco-maven-plugin:0.8.1:check (default-check)
Analyzed bundle 'checkstyle' with 520 classes
````.

**Comment by romani** [comment]
problem is reproduced on my local:
```
$ mvn verify
....
[INFO] --- jacoco-maven-plugin:0.8.1:check (default-check) @ checkstyle ---
[INFO] Loading execution data file /home/rivanov/java/github/romani/checkstyle/target/jacoco.exec
[INFO] Analyzed bundle 'checkstyle' with 520 classes
[WARNING] Rule violated for class com.puppycrawl.tools.checkstyle.checks.naming.ParameterNameCheck:
branches covered ratio is 0.96, but expected minimum is 1.00
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
```

but:
```
$ mvn -e clean test jacoco:restore-instrumented-classes jacoco:report@default-report
....
[INFO] --- jacoco-maven-plugin:0.8.1:report (default-report) @ checkstyle ---
[INFO] Loading execution data file /home/rivanov/java/github/romani/checkstyle/target/jacoco.exec
[INFO] Analyzed bundle 'checkstyle' with 396 classes
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS

```

**Comment by kazachka** [comment]
Updated

**Comment by romani** [comment]
still coverage problem, please run on local `./.ci/travis/travis.sh jacoco`

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheck.java [hunk 1]
```diff
@@ -187,7 +191,8 @@ protected boolean mustCheckName(DetailAST ast) {
         if (ignoreOverridden && isOverriddenMethod(ast)
                 || ast.getParent().getType() == TokenTypes.LITERAL_CATCH
                 || CheckUtil.isReceiverParameter(ast)
-                || !matchAccessModifiers(getAccessModifier(ast))) {
+                || !matchAccessModifiers(getAccessModifier(ast))
+                || ast.getParent().getType() == TokenTypes.LAMBDA) {
```
[the remark this answers is not in the document]
Comment (kazachka) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheck.java]: It didn't help

[the remark this answers is not in the document]
Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheck.java]: I was just saying to recheck that coverage will still pass once you fix the initial problem. Sometimes changing conditional orders changes code coverage too.

**Comment by rnveach** [comment]
@kazachka Are you able to fix CI?

**Comment by kazachka** [comment]
I've ran travis on my local and it's showed 100% coverage.
Result is here:
https://kazachka.github.io/parameter-check-name-jacoco/

**Comment by rnveach** [comment]
@kazachka 
https://kazachka.github.io/parameter-check-name-jacoco/com.puppycrawl.tools.checkstyle.checks.naming/ParameterNameCheck.java.html#L238
This shows no changes you made in the check. I assume you aren't using the correct branch when you ran this.

**Comment by kazachka** [comment]
Sorry, it was another branch.
Updated report and it showed strange results. Coverage missed on code, which wasn't change.
https://kazachka.github.io/parameter-check-name-jacoco/

**Comment by rnveach** [comment]
Yes, missed coverage is because the code before the call to `getAccessModifier` changed, and this affected everything after it. This shows there are no tests that use specific access modifiers on constructors.
This must have only been being hit before because our cases were all lambdas.

Please fix coverage by expanding tests.

**Comment by kazachka** [comment]
It looks like there only METHOD_DEF and CTOR_DEF can be parents of PARAMETER_DEF checked by ParameterName. So when it checks meth.getType() == CTOR_DEF it's always true.

**Comment by kazachka** [comment]
Constructor modifiers checks here
https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/parametername/InputParameterNameAccessModifier.java#L5

**Comment by romani** [comment]
@kazachka , take a look at history of this method, it such condition was added when lambda was added, so it is ok to change it.

Please update code to be as required, to be always covered and run regression diff testing, any changes will show a case that our tests miss.

**Comment by kazachka** [comment]
Updated

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheckTest.java [hunk 2]
```diff
@@ -174,4 +174,11 @@ public void testReceiverParameter() throws Exception {
         verify(checkConfig, getPath("InputParameterNameReceiver.java"), expected);
     }
 
+    @Test
+    public void testLambdaParameter() throws Exception {
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheckTest.java]: testLambdaParameter ==> testLambdaParameterNoViolationAtAll

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/parametername/InputParameterNameLambda.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package com.puppycrawl.tools.checkstyle.checks.naming.parametername;
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+public class InputParameterNameLambda {
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/parametername/InputParameterNameLambda.java]: please add multiline comment above to make it clear that no violations are expected, as lambda is covered by  another Check.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheck.java [hunk 4]
```diff
@@ -202,19 +207,16 @@ protected boolean mustCheckName(DetailAST ast) {
      * @return the access modifier of the method/constructor.
      */
     private static AccessModifier getAccessModifier(final DetailAST ast) {
-        final DetailAST params = ast.getParent();
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheck.java]: Check is changed in non simple way - we need do regression diff testing, please run on all projects.
There will be too much differences if we run them in simple way as we do usually.
So we need to make master config with only one Check ParameterNameCheck , Patch config with two Checks (ParameterNameCheck, LambdaParameterName) to make sure that after change violations are stay as before, the only difference is name of Check in violation.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheckTest.java (hunk 2 above)
Comment (kazachka) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheckTest.java]: Done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/parametername/InputParameterNameLambda.java (hunk 3 above)
Comment (kazachka) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/parametername/InputParameterNameLambda.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheck.java (hunk 4 above)
Comment (kazachka) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheck.java]: Done. Actually there no violations on projects except of Vavr.
https://kazachka.github.io/parameter-name-check-diff/

**Pull request merged**